### PR TITLE
Try to properly handle `get_node` failures. Fixes #386.

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

File: saltcloud/clouds/libcloud_aws.py [hunk 1]
```diff
@@ -284,6 +298,9 @@ def create(vm_):
                     key_filename=__opts__['AWS.private_key']):
                 username = user
                 break
+    if data.public_ips:
```
Comment (techhat) [inline saltcloud/clouds/libcloud_aws.py]: Was this properly placed? It doesn't look right.

Comment (s0undt3ch) [inline saltcloud/clouds/libcloud_aws.py]: we need to break out of the while True loop. I'll do some more testing when I get home.

Comment (techhat) [inline saltcloud/clouds/libcloud_aws.py]: That's what I figured, but that's also not where it is.

**Comment by s0undt3ch** [comment]
Should be good now ;)

**Pull request merged**